File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> v1(26,0) ;
       // vector <int> v2(26,0) ;
        if(s.size()!=t.size())
        return false ;

        for(int i = 0 ;  i<s.size() ;i++)
        {
            v1[s[i]-'a']++ ;
            v1[t[i]-'a']-- ;
        }

        for(int  i : v1)
        {
            if(i!=0)
            return false  ;
        }

        return true;
    }
};

File: Data Structures & Algorithms/reorder-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:

        fast  = head 
        slaw = head
        l1 = head
        while fast and fast.next:
            slaw = slaw.next
            fast  = fast.next.next
        
        l2 = slaw.next
        slaw.next =None
       
        prev = None
        nxt = None
        while l2:
            nxt = l2.next 
            l2.next = prev
            prev = l2
            l2=nxt
        r = head
        while prev:
            t1 = r.next 
            t2 = prev.next
            r.next = prev
            prev.next = t1 
            prev = t2
            r = t1 
    

